internals/secrethub/init.go

```diff
@@ -105,12 +121,98 @@ func (cmd *InitCommand) Run() error {
 	case InitModeSignup:
 		signupCommand := SignUpCommand{
 			io:              cmd.io,
-			newClient:       cmd.newClient,
+			newClient:       cmd.newUnauthenticatedClient,
 			credentialStore: cmd.credentialStore,
 			progressPrinter: cmd.progressPrinter,
 			force:           cmd.force,
 		}
 		return signupCommand.Run()
+	case InitModeSetupCode:
+		setupCode := cmd.setupCode
+		if setupCode == "" {
+			var err error
+			setupCode, err = ui.AskAndValidate(cmd.io, "What is your setup code?\n", 3, api.ValidateSetupCode)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprintf(
+			cmd.io.Output(),
+			"An account credential will be generated and stored at %s. "+
+				"Losing this credential means you lose the ability to decrypt your secrets. "+
+				"So keep it safe.\n",
+			credentialPath,
+		)
+
+		// Only prompt for a passphrase when the user hasn't used --force.
+		// Otherwise, we assume the passphrase was intentionally not
+		// configured to output a plaintext credential.
+		var passphrase string
+		if !cmd.credentialStore.IsPassphraseSet() && !cmd.force {
+			var err error
+			passphrase, err = ui.AskPassphrase(cmd.io, "Please enter a passphrase to protect your local credential (leave empty for no passphrase): ", "Enter the same passphrase again: ", 3)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprint(cmd.io.Output(), "Setting up your account...")
+		cmd.progressPrinter.Start()
+
+		client, err := cmd.newClientWithSetupCode(setupCode)
+		if err != nil {
+			return err
+		}
+
+		credential := credentials.CreateKey()
+		_, err = client.Credentials().Create(credential, "")
+		if err != nil {
+			return err
+		}
+
+		exportKey := credential.Key
+		if passphrase != "" {
+			exportKey = exportKey.Passphrase(credentials.FromString(passphrase))
+		}
+
+		encodedCredential, err := credential.Export()
+		if err != nil {
+			cmd.progressPrinter.Stop()
+			return err
+		}
+
+		err = cmd.credentialStore.ConfigDir().Credential().Write(encodedCredential)
+		if err != nil {
+			cmd.progressPrinter.Stop()
+			return err
+		}
+
+		client, err = cmd.newUnauthenticatedClient()
```

Comment:
I can't come up with any way in which creating an unauthenticated client doesn't work, as it will try to load the credential from its default spots, but I believe it's safer to create a client already with the credential:

```suggestion
		client, err = cmd.newClientWithCredentials(credential)
```

It's certainly more explicit (and therefore easier to understand). And I believe it can also prevent an unneeded password prompt (and certainly prevents unneeded reading and decrypting).